src/reducers/computeRenderedRows.js

```diff
@@ -239,3 +245,97 @@ function computeRenderedRowOffsets(state, rowRange) {
   state.rowOffsets = rowOffsetsCache;
   state.rows = bufferMapping;
 }
+
+
+/**
+ * Walk the rows to render and compute the height offsets and
+ * positions in the row buffer.
+ *
+ * NOTE (jordan) This alters state so it shouldn't be called
+ * without state having been cloned first.
+ *
+ * @param {!Object} state
+ * @param {{
+ *   endBufferIdx: number,
+ *   endViewportIdx: number,
+ *   firstBufferIdx: number,
+ *   firstViewportIdx: number,
+ * }} rowRange
+ * @private
+ */
+function computeRenderedRowOffsetsInViewport(state, rowRange) {
```

Comment:
Is there any opportunity to deduplicate this method w/ computeRenderedRowOffsets.  They are very similar.

Reply:
This was done in #403. It had some slight problems with readability, but we can try and combine this again after computeRenderedRowOffsetsInViewPort is done.

Reply:
We'll do this on the follow up PR